msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/msalonly/basic/TestCase532736.java

```diff
@@ -0,0 +1,116 @@
+import androidx.test.uiautomator.UiSelector;
+
+import com.microsoft.identity.client.Prompt;
+import com.microsoft.identity.client.msal.automationapp.AbstractMsalUiTest;
+import com.microsoft.identity.client.msal.automationapp.R;
+import com.microsoft.identity.client.msal.automationapp.sdk.MsalAuthTestParams;
+import com.microsoft.identity.client.msal.automationapp.sdk.MsalSdk;
+import com.microsoft.identity.client.ui.automation.TokenRequestTimeout;
+import com.microsoft.identity.client.ui.automation.browser.BrowserChrome;
+import com.microsoft.identity.client.ui.automation.device.settings.ISettings;
+import com.microsoft.identity.client.ui.automation.interaction.OnInteractionRequired;
+import com.microsoft.identity.client.ui.automation.utils.CommonUtils;
+import com.microsoft.identity.client.ui.automation.utils.UiAutomatorUtils;
+import com.microsoft.identity.labapi.utilities.client.LabQuery;
+import com.microsoft.identity.labapi.utilities.constants.AzureEnvironment;
+import com.microsoft.identity.labapi.utilities.constants.TempUserType;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+// MSAL Falls Back on WebView When All Browsers are Disabled
+// https://identitydivision.visualstudio.com/DevEx/_workitems/edit/532736
+public class TestCase532736 extends AbstractMsalUiTest {
+
+    @Test
+    public void test_532736() throws Throwable {
+        final String username = mLabAccount.getUsername();
+
+        // Disable Chrome
+        ISettings settings = getSettingsScreen();
+        BrowserChrome chrome = ((BrowserChrome) mBrowser);
+
+        settings.disableAppThroughSettings(chrome.getPackageName());
+
+        final MsalSdk msalSdk = new MsalSdk();
+
+        final MsalAuthTestParams authTestParams = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .loginHint(username)
+                .scopes(Arrays.asList(mScopes))
+                .promptParameter(Prompt.SELECT_ACCOUNT)
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+
+        msalSdk.acquireTokenInteractive(authTestParams, new OnInteractionRequired() {
+            @Override
+            public void handleUserInteraction() {
+                // Assert that webview UI object exists
+                final UiObject webView = UiAutomatorUtils.obtainUiObjectWithUiSelector(new UiSelector().className(WebView.class), CommonUtils.FIND_UI_ELEMENT_TIMEOUT);
+                Assert.assertTrue(webView.exists());
+
+                // Assert that no objects from Chrome package are present
+                final UiObject chromeObject = UiAutomatorUtils.obtainUiObjectWithUiSelector(new UiSelector().packageName("com.android.chrome"), CommonUtils.FIND_UI_ELEMENT_TIMEOUT);
+                Assert.assertFalse(chromeObject.exists());
+
+                // Some cleanup after the test passes
+                settings.enableAppThroughSettings(chrome.getPackageName());
```

Comment:
Should this be happening in an `after` block? Perhaps in a test rule?

Reply:
Yes this would make more sense. I'm going to do an @after block because this is only used in a single case.